peakina/readers/excel.py

```diff
@@ -0,0 +1,133 @@
+"""
+Module to add xml support
+"""
+import enum
+from io import StringIO
+from typing import Any, List, Optional
+
+import pandas as pd
+import xlrd
+from openpyxl import load_workbook
+from openpyxl.utils.exceptions import InvalidFileException
+
+from peakina.readers.csv import PreviewArgs
+
+
+class EXCEL_TYPE(enum.Enum):
+    NEW = "new"
+    OLD = "old"
+
+
+def _read_old_xls_format(wb: Any, sh_name: str, preview: Optional[PreviewArgs]) -> Any:
+    """ """
+    if preview:
+        to_iter = range(
+            preview.get("offset", 0), preview.get("offset", 0) + preview.get("nrows", 500) + 1
+        )
+    else:
+        to_iter = range(wb[sh_name].nrows)
+
+    for rx in to_iter:
+        yield wb[sh_name].row(rx)
+
+
+def _read_new_xls_format(wb: Any, sh_name: str, preview: Optional[PreviewArgs]) -> Any:
+    """ """
+
+    return wb[sh_name].iter_rows(
+        min_row=preview.get("offset", 0) if preview else None,
+        max_row=preview.get("offset", 0) + preview.get("nrows", 500) if preview else None,
+        values_only=True,
+    )
+
+
+def _get_row_to_iterate(
```

Comment:
```suggestion
def _rows_iterator
```
or `_get_rows_iterator` or something like this

Reply:
👍🏾 Updated !